.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pycqa/flake8
+    rev: '4.0.1'
+    hooks:
+    -   id: flake8
+        stages: [push]
```

Comment:
I actually liked the fact of running flake8 at every commit more.

The point of having pre-commit is to avoid "fix flake8" (or fix mypy") commits. 
if we only run it on push - we'll simply end up with that commit at the end - not removed.